### DISCUSSION TITLE
Send broadcast message with rabbitmq reactive messaging

### DISCUSSION THREAD

**Discussion by iompo** [quarkusio/quarkus#31322] [Q&A] [answered]
I'm trying to send a broadcast message using the rabbitmq reactive library. I tried to follow the [documentation](https://quarkus.io/guides/rabbitmq-reference#incoming-channel-configuration) and define an incoming broadcast channel, but the setting seems to be ignored.
Looking at the RabbitMQ [publish/subscriber documentation](https://www.rabbitmq.com/tutorials/tutorial-three-java.html) broadcasting a message requires temporary queue with random names or names auto-generated from the server.
Quarkus requires the queue name to be specified, so what is the suggested way to define a temporary queue?

[a comment by an automated account is not shown]

**Comment by ozangunalp** [comment, marked as answer]
The `broadcast` config for an incoming channel in reactive messaging corresponds to how consumed messages are dispatched to different channels. 

For a rabbitmq pub/sub setup you'd need to publish to a fanout exchange and bind consumer queues to that exchange.

**Reply by ozangunalp** [reply]
If we follow the example from the RabbitMQ tutorial you linked, 
for producing to logs exchange:
`mp.messaging.outgoing.sink.exchange.name=logs`
`mp.messaging.outgoing.sink.exchange.type=fanout`
and in each of your consumers:
`mp.messaging.incoming.in.queue.name=(server.auto)`
`mp.messaging.incoming.in.exchange.name=logs`
`mp.messaging.incoming.in.exchange.type=fanout`

The declaration and binding will be done automatically. With `(server.auto)` for queue name declared queue will have a generated name by the broker

**Reply by iompo** [reply]
thanks! I couldn't find that `(server.auto)` anywhere

**Reply by ozangunalp** [reply]
I realize that it is not documented. And also not coherent with the default exchange value that is `""`. We align those two.
